llvm/lib/SYCLLowerIR/LocalAccessorToSharedMemory.cpp

```diff
@@ -14,92 +14,115 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "LocalAccessorToSharedMemory.h"
-#include "../MCTargetDesc/NVPTXBaseInfo.h"
+#include "llvm/SYCLLowerIR/LocalAccessorToSharedMemory.h"
 #include "llvm/IR/GlobalValue.h"
 #include "llvm/IR/Instructions.h"
 #include "llvm/IR/PassManager.h"
+#include "llvm/Support/CommandLine.h"
 #include "llvm/Transforms/IPO.h"
 
 using namespace llvm;
 
 #define DEBUG_TYPE "localaccessortosharedmemory"
 
+static bool EnableLocalAccessor;
+
+static cl::opt<bool, true> EnableLocalAccessorFlag(
+    "sycl-enable-local-accessor", cl::Hidden,
+    cl::desc("Enable local accessor to shared memory optimisation."),
+    cl::location(EnableLocalAccessor), cl::init(false));
+
 namespace llvm {
 void initializeLocalAccessorToSharedMemoryPass(PassRegistry &);
-}
+} // namespace llvm
 
 namespace {
 
 class LocalAccessorToSharedMemory : public ModulePass {
```

Comment:
Should it be supported by new Pass Manager?

Reply:
I'm not sure, feels like a design/policy issue. FWIW passes both here (`SYCLLowerIR`) and in the AMDGPU/NVPTX backends still use legacy.

I'll be happy to move it to the new PM, but perhaps in a follow up?

Reply:
Ok, I don't mind if it would be moved to the new PM in a separate PR. Thank you.

Reply:
Here is an example of porting this pass to new PM interface: https://github.com/intel/llvm/pull/3264.

Reply:
I've created an issue for the porting (https://github.com/intel/llvm/issues/5310) and added it to the backlog.